srtcore/api.cpp

```diff
@@ -344,7 +350,10 @@ SRTSOCKET CUDTUnited::generateSocketID()
             bool exists = false;
             {
                 CGuard cg(m_GlobControlLock);
-                exists = m_Sockets.count(sockval);
+                exists = for_group ?
+                    m_Groups.count(sockval | SRTGROUP_MASK)
+                 :
+                    m_Sockets.count(sockval);
             }
```

Comment:
The handing curly braces {} (code block) are treated by SonarCloud as a technical dept, as the internals of the block should become a function.
The following way is better, as you get  `exists` constant and it is more readable.
```c++
m_GlobControlLock.lock();
const bool exists = for_group
    ? m_Groups.count(sockval | SRTGROUP_MASK)
    : m_Sockets.count(sockval);
m_GlobControlLock.unlock();
```

Reply:
I'd prefer enterCS/leaveCS, beside that, good idea.